cloudify_awssdk/autoscaling/resources/launch_configuration.py

```diff
@@ -106,6 +107,12 @@ def create(ctx, iface, resource_config, **_):
     params[RESOURCE_NAME] = resource_id
     utils.update_resource_id(ctx.instance, resource_id)
 
+    # Check if the "IamInstanceProfile" is passed or not and then update it
+    iam_instance_profile = params.get(IAM_INSTANCE_PROFILE)
+    if iam_instance_profile and isinstance(iam_instance_profile, basestring):
+        iam_instance_profile = iam_instance_profile.strip()
+        params[IAM_INSTANCE_PROFILE] = str(iam_instance_profile)
```

Comment:
What if we have iam_instance_profile that is not a basestring? What is the case?

Reply:
Simply it will failed when run the installation for the blueprint, I will add an else to raise `NonRecoverableError`